Data/HashMap/Internal/Array.hs

```diff
@@ -190,7 +194,12 @@ instance Show a => Show (Array a) where
 -- Array wrappers, but it's still slightly bogus.
 unsafeSameArray :: Array a -> Array b -> Bool
 unsafeSameArray (Array xs) (Array ys) =
-  tagToEnum# (unsafeCoerce# reallyUnsafePtrEquality# xs ys)
+  tagToEnum#
+#if __GLASGOW_HASKELL__ >= 903
+    (reallyUnsafePtrEquality# xs ys)
+#else
+    (unsafeCoerce# reallyUnsafePtrEquality# xs ys)
+#endif
```

Comment:
@doyougnu Sorry for the late request, but could you move these changes to a separate PR? They seem rather orthogonal to the base change.

Reply:
@sjakobi just rebased and dropped that commit. I'm unsure if another PR is needed for that one, its just applying the unordered-containers patch from head.hackage/patches.

Reply:
Thanks! It seems related to https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.4#reallyunsafeptrequality. We can add the patch now or alternatively wait until there's a release candidate and we're a bit more sure that the patch is actually necessary.